Eirin's Trial task

theproactivedev opened the pull request:
**#1 Task**
I'm confused with this one but what I just did is I removed the html tag in base.scss. I'm not sure if I should check all scss files imported from base.scss and reduce selectors there.

**#2 Task**
_**How will you go about ensuring your change is working?**_
I use the browser to resize the screen while looking at the pixel number at the top right of the screen or select a device in the browser developer tools. And I also use mobile phones, tablets, small screens and bigger screens to check if styles are being applied to a specific media query.

_**Do you have a preference over units used in media queries? Explain why.**_
I can't say I have a preference but I'm comfortable with pixels. I always use pixels in media queries because it's working consistently in all browsers even when the user zooms in. There are articles showing experiments on px, rem and em on which is the best unit to use in media queries. But in the end, I think it should be what is working consistently on all browsers.

**#3 Task**
**_Why is this part updated?_**
**_What have you updated?_**
- I added a main tag because user needs to have a landmark for the main content. I also added styles for main tag in home/home.scss to center its child elements.
- I changed some icons to be button so I can tab through them. Or if I'll use the div (the element before), I can just add a tabindex attribute.
- Then, I changed child divs of main element into sections because that whole section directly portray a topic.
- I also added aria-labels on anchor tags, icon buttons, input elements and select element so screen readers can recognize it.
- I also changed `aside` elements to `div` elements because some content are not considered as secondary, more like, they are actually adding value to the main content. And some content inside aside are not supporting a primary content like the social icons.
- I also changed duplicate id values because id should be unique in one page.
- I made the aside element with id of `latest-posts-navigation` to a `nav` because it's not adding a secondary content. It's more of a navigation just like what it says in the class name.
- I also wanted to add an article element in the div element with class name, `featured-and-related-posts` so I can separate the section label and the title of the article. Then, I'll just change some styles to fix the issue.
- I removed excess closing `p` tags which should not be actually there.
- And I replaced some span elements to be a `p` element because the content inside actually has meaning.

_**What tools did you use to find issues here?**_
I used tools such as Lighthouse, Axe Accessibility and W3C validator to check semantics of the markup. I did/change these things based on the result, me tabbing through the content and my personal feeling or understanding on the content.

Thank you! Let me know your feedback about this. I'll appreciate it.

gkweb commented on src/scss/base.scss:
```diff
@@ -20,54 +20,52 @@
-		display: flex;
-		flex-direction: column;
-		align-items: center;
+	display: flex;
```
What are your thoughts regarding the use of `display: flex` here?

theproactivedev replied:
If I'm going to remove my main element, I think it's a quick and smart fix on centering all child elements having different widths. However, I feel like I will still add my main tag. And I will add styles or add a class on the main navigation, footer and main tag to center their content. I feel that the `display:flex` on the body tag made adding styles a little more complicated.

theproactivedev commented:
#4 Task

I am not comfortable with overflow-x hidden. If I'm a reader or a traveler who wants details and the content I'm reading gets clipped and is invisible, I don't feel good about that at all. I didn't get what I want or need from the website. So I'll just look for another website with the travel details I need. But if I can still read the content even though there's an unexpected scrollbar, that's okay as long as I can read the content and get details.

As a developer, we can just set width as percentage values to elements, make use of box-sizing, use the media breakpoints and test the styles to check content for overflowing and to deliver a UI that is not just beautiful but also meaningful and useful. 

If the variant is a margin, we add left and right margin with a value of `horizontal`. Based on the values and notes in the responsive-spacing.scss, horizontal is `var(--gutter-24)` and `var(--gutter-16)`. And gutter is meant for horizontal spacing.

Since margin is not included in the box-sizing, we have to deduct these left and right margins from the width and max-width.

Currently, we're deducting `var(--spacer-24)` and `var(--spacer-16)` that refers to a vertical spacing and we aren't even sure that these refers to the `horizontal` values we added as left and right margins earlier in the code. 

Although `var(--gutter-24)` is equal to `var(--spacer-24)` and `var(--gutter-16)` is equal to `var(--spacer-16)`, it might lead to confusion for the coming developers.

I suggest we use `var(--gutter-24)` and `var(--gutter-16)` which are meant for horizontal spacing to deduct these from the width and max-width. (I also want to somehow show to coming developer that the horizontal value we're passing to responsive-spacing is equal to  `var(--gutter-24)` and `var(--gutter-16)` )

And I also added `Header` because Header is currently using the variant, margin.

#5 Task

I feel like overflow-y:hidden is useless here. Should the header panel have a fixed height? And are we hiding something under header panel that we have to make overflow-y hidden? 

If we add the overflow-x hidden and overflow-y hidden, content will be clipped. And users will get annoyed. Isn't that more annoying than a scrollbar that only appears when needed?

Then I also changed em to rem because using em means being extra careful on declaring font size. And if we use em, new developers might have a hard time setting font-size because he will have to know (or look for the font-size declaration) what font-size is declared to the parent element to prevent unexpected big or small font-size. If we use rem, we have single basis (or single source of truth. hehe.) for our spaces and font sizes.

Then, I also changed large, small, smaller to large, medium, small because I feel that's easier for new developers who'll code here and I think that's more common even to our things like t-shirt, ordering food at the fastfood. (I'm also thinking but not really sure that maybe we can explicitly show that large, medium and small are for vertical spacing. Horizontal seems unrelated to large, medium, small.)

Those are my thoughts. Thank you for this trial task/interview.

@gkweb Let me know your feedback. Thank you!

gkweb commented on src/scss/container/container.scss:
```diff
@@ -34,15 +34,16 @@
 
 		@include media($small...) {
-			width: calc(100% - var(--spacer-16) * 2);
+			width: calc(100% - var(--gutter-16) * 2);
```
👍

gkweb commented on src/scss/home/home.scss:
```diff
@@ -11,6 +11,13 @@
 @import '../plastic-transition/transition.scss';
 
 @mixin home {
+	main {
```
👍

gkweb commented on src/scss/responsive-spacing/responsive-spacing.scss:
```diff
@@ -18,22 +18,22 @@
-		--spacer-32: 2em;
-		--gutter-16: 1em;
-		--gutter-24: 1.5em;
+		--spacer-4: 0.25rem;
```
💯

theproactivedev commented:
@gkweb That's good to know! :D Thank you for your time.
